src/components/DataProvider/index.js

```diff
@@ -78,33 +105,32 @@ export default class DataProvider extends Component {
         baseDomain,
         subDomain,
         pointsPerSeries,
-        oldSeries: s,
+        oldSerie: s,
         reason,
       });
+      const data =
+        Array.isArray(loaderReturn.data) && loaderReturn.data.length
+          ? loaderReturn.data
+          : [];
       return {
-        data: [],
+        ...s,
```

Comment:
This is breaking a lot of the other stories. The reason why `this.getSeriesObjects` are used in `render` as opposed to in `fetchData` is to allow updates to propagate when the `series` props change (`zoomable`, `hidden` etc). Now -- nothing will happen until `fetchData` is called again.

Reply:
Longer explanation;

The thought behind `getSeriesObjects` is to merge the two different series configs. One set of properties comes from the `series` prop of `DataProvider` (`id`, `loaders`, `zoomable` etc) and the other set of properties comes from the loader itself (`data`, `step`, `drawPoints`). These two configs can be updated independently, so merging them together in `render` ensures that whichever one of those changed, they get merged and shipped through the `context`. If we update the state only in `fetchData` then the `series` prop sent through `DataProvider` will not reflect any changes until `fetchData` is called and `enhancedSeries` is re-built.